File: ValidParentheses/ValidParentheses.cpp
#include<string>
#include<stack>

using namespace std;

class Solution {
public:
	bool isValid(string s) {
		stack<char> tmp;
		for (char &c : s) {
			if (c == '(' || c == '[' || c == '{')
				tmp.push(c);
			else if (c == ')') {
				if (tmp.empty() || tmp.top() != '(')
					return false;
				else
					tmp.pop();
			}
			else if (c == ']') {
				if (tmp.empty() || tmp.top() != '[')
					return false;
				else
					tmp.pop();
			}
			else {
				if (tmp.empty() || tmp.top() != '{')
					return false;
				else
					tmp.pop();
			}
		}
		return tmp.empty();
	}
};

File: ValidParentheses/ValidParentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        tmp = []
        dic = {")":"(", "]":"[", "}":"{"}
        for i in s:
            if i in dic.values():
                tmp.append(i)
            elif i in dic.keys():
                if tmp == [] or dic[i] != tmp.pop():
                    return False
            else:
                return False
        return tmp == []